### DISCUSSION TITLE
Sending email with attachment is failing when using awsbatch executor

### DISCUSSION THREAD

**Discussion by mp-seqera** [nextflow-io/nextflow#3332] [Q&A]
I've tested a couple of different versions of sending email but usually still ends up having an error when using awsbatch as executor for both DSL1 and DSL2. Below is the debug and trace logs of the most recent testing using the latest nextflow version. This works when tested with Slurm.

```
EDIT: Updated the logs from today's run using 22.10.0


Oct.-31 18:15:33.186 [Actor Thread 9] ERROR nextflow.extension.OperatorImpl - @unknown
javax.mail.MessagingException: The following attachment file does not exist: /nf-mpanganiban/scratch/f0/4564d7c7a4c32bc379973160d719fb/ASD_output.txt
	at nextflow.mail.Mailer.createAttachment(Mailer.groovy:339)
	at nextflow.mail.Mailer.createMimeMessage(Mailer.groovy:328)
	at nextflow.mail.Mailer.send(Mailer.groovy:421)
	at nextflow.mail.Mailer.send(Mailer.groovy:496)
```

Replication:
```
#!/usr/bin/env nextflow

nextflow.enable.dsl=2

process generate_email {
    debug true
    input:
        val(input_file_content)

    output:
        path "*_output.txt"

    script:
        """
        echo "${input_file_content}" > sample_output.txt
        """
}

workflow { 
    main: 
        channel.from("Here is some text")
        | generate_email
        | flatten
        | map { file_attachment ->
             sendMail {
                 to 'email@email.com'
                 attach "${file_attachment}"
                 subject "TEST EMAIL"
                 """
                 TEST CONTENT
                 """
             }
        }
}
```
Full Logs: [nextflow_s3.log](https://github.com/nextflow-io/nextflow/files/9905672/nextflow_s3.log)

**Comment by pditommaso** [comment]
please include the full .nextflow.log file

**Comment by pditommaso** [comment]
and @robsyme knows the cause of this

**Comment by mp-seqera** [comment]
Full Logs: [nextflow_s3.log](https://github.com/nextflow-io/nextflow/files/9905672/nextflow_s3.log)

**Comment by pditommaso** [comment]
The problem is that the attachment file path is wrapped into a string 

```
     attach "${file_attachment}"
```

it should be 

```
   attach file_attachment
```

Files are not strings!

**Reply by mp-seqera** [reply]
Here is the update error when using it as a file.
```
Nov.-01 06:06:20.781 [Task monitor] TRACE nextflow.processor.TaskProcessor - Finalize process > generate_report (1)
Nov.-01 06:06:20.782 [Task monitor] TRACE nextflow.processor.TaskProcessor - Process generate_report > collecting out param: fileoutparam<0> = /nf-mpanganiban/scratch/ce/fa3c8ccde1bbe0097c519080df35fc
/ASD_output.txt
Nov.-01 06:06:20.782 [Task monitor] TRACE nextflow.processor.TaskProcessor - Process generate_report > Binding out param: fileoutparam<0> = [/nf-mpanganiban/scratch/ce/fa3c8ccde1bbe0097c519080df35fc/A
SD_output.txt]
Nov.-01 06:06:20.783 [Task monitor] TRACE nextflow.processor.TaskProcessor - <generate_report> Binding param fileoutparam<0> with [/nf-mpanganiban/scratch/ce/fa3c8ccde1bbe0097c519080df35fc/ASD_output.
txt]
Nov.-01 06:06:20.805 [Actor Thread 9] ERROR nextflow.extension.OperatorImpl - @unknown
java.lang.UnsupportedOperationException: null
        at com.upplication.s3fs.S3Path.toFile(S3Path.java:452)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)

```

Full logs: 
[nextflow_file.log](https://github.com/nextflow-io/nextflow/files/9909889/nextflow_file.log)

**Reply by pditommaso** [reply]
Now it makes sense. Essentially S3 files are remote objects, they cannot used for email attachments

**Reply by robsyme** [reply]
Thanks Paolo, Mark. 
Can I confirm that (at least for the moment), sending emails with attachments that are S3 objects is not supported? Is this something that we'd like to make available as a feature?

**Reply by pditommaso** [reply]
in principle, it should not be to too hard. It should be checked if it's a remote object and copy to a tmp `File`. But I'm already expecting users to create several gigas attachments 😄

**Reply by robsyme** [reply]
From what I've seen, people are being very sensible with their email attachment sizes. I'd hope that the email server will bounce anything unreasonable.
Would it be helpful if I wrote this up as an issue with the `kind/feature` tag?

**Reply by pditommaso** [reply]
agreed

**Comment by robsyme** [comment]
Feature request: https://github.com/nextflow-io/nextflow/issues/3349
